app/uk/gov/hmrc/emailverification/services/JourneyService.scala

```diff
@@ -113,7 +129,7 @@ class JourneyService @Inject() (
           case Some(email) => verificationStatusRepository.lock(journey.credId, email).map(_ => result)
           case None        => Future.successful(result)
         }
-      case Some(journey) if journey.passcodesSentToEmail > config.maxAttemptsPerEmail =>
+      case Some(journey) if journey.passcodesSentToEmail >= config.maxAttemptsPerEmail => // [GG-6678] recordPasscodeResent increments but does not return an incremented value
```

Comment:
in the JourneyService, the method "resendPasscode" first action is to call the journeyRepository.recordPasscodeResent method, which increments "passcodesSentToEmail" by 1 in the repository and returns a journey. However, after testing i discovered recordPasscodeResent does increment the value but returns a journey with the old passcodesSentToEmail value NOT the new value. This was misleading and caused the code to be written thinking that recordPasscodeResent was returning the incremented value.

Originally in the method there are two comparisons to config values with one set to ">=" (because the value is not incremented) and ">" (because the value is incremented). I had to change the ">" to ">=" because the code doesn't return the journey with the newly incremented value. I wrote the JourneyRepositorySpec.scala test to show that the method returns the journey with the old value while simultaneously incrementing the value. I also left a comment so people are aware of this behaviour, as it looks like to me it was changed previously with the misunderstanding of this unique characteristic.

Reply:
ok, thx